firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigSharedPrefsClient.java

```diff
@@ -249,6 +257,46 @@ void setBackoffMetadata(int numFailedFetches, Date backoffEndTime) {
     }
   }
 
+  public void setCustomSignals(Map<String, Object> newCustomSignals) {
+    synchronized (customSignalsLock) {
```

Comment:
Do we know how frequent are we trigger this setter? This setter involves read from SharedPreferences, doing JSON serialization and update SharedPreferences which could be quiet expensive. 

The executor which we are using is the common blocking executor which is shared across **all** firebase SDKs. Reference can be found [here](https://firebase.github.io/firebase-android-sdk/components/executors/#blocking). This is mean all firebase SDK will share the 4 threads which managed by the blocking executor and we don't want any SDK over use them. 

Can we implement some validation logic to minimize the potential of polluting the common executor? For example, check if signal is getting manipulated and if not let's not proceed this expensive process. Ideally the logic should be you save an instance of signal in memory and keep updating on that. Whenever the in memory instance does not match with on-disk record you write to disk. We have this logic in Crashlytics for updating custom key value. [Here is the reference](https://github.com/firebase/firebase-android-sdk/blob/main/firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/UserMetadata.java#L242)

Reply:
While we don't expect frequent calls to this setter, you're right that excessive use could be costly. To mitigate this, I've implemented a check to ensure the signal has actually been modified before updating. This avoids unnecessary work and reduces the load on the shared executor. Similar to the Crashlytics custom key implementation, the setter now compares the in-memory signal instance with the on-disk record and only updates when they differ